buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/CiReportingPlugin.kt

```diff
@@ -29,34 +32,75 @@ open class CiReportingPlugin : Plugin<Project> {
     override fun apply(project: Project): Unit = project.run {
         if (BuildEnvironment.isCiServer) {
             gradle.buildFinished {
-                prepareReportsForCiPublishing()
+                val failedTasks = project.failedTasks()
+                val executedTasks = project.executedTasks()
+                val tmpTestFiles = project.subprojects.flatMap { it.tmpTestFiles() }
+                prepareReportsForCiPublishing(failedTasks, executedTasks, tmpTestFiles)
+                verifyTestFilesCleanup(failedTasks, tmpTestFiles)
             }
         }
     }
 
     private
-    fun Project.prepareReportsForCiPublishing() {
-        val failedTaskGenericHtmlReports = failedTasks(allprojects).flatMap {
-            it.failedTaskGenericHtmlReports()
-        }
-        val attachedReports = executedTasks(subprojects).flatMap {
-            it.attachedReportLocations()
-        }
-        val failedTaskCustomReports = failedTasks(subprojects).flatMap {
-            it.failedTaskCustomReports()
+    fun verifyTestFilesCleanup(failedTasks: List<Task>, tmpTestFiles: List<Pair<File, String>>) {
+        if (failedTasks.none { it is Test }) {
+            if (tmpTestFiles.isNotEmpty()) {
```

Comment:
We should really use the same logic as we do in the `TestFilesCleanUpPlugin` and in `EmptyDirectoryCheck`:
- respect the `testFilesCleanup` extension setting (per project)
- write the report and create the same message as the `EmptyDirectoryCheck`.

The the test files cleanup plugin and `EmptyDirectoryCheck` can be removed.